rust/arrow/src/array.rs

```diff
@@ -609,8 +609,8 @@ impl From<ArrayDataRef> for BinaryArray {
 
 impl<'a> From<Vec<&'a str>> for BinaryArray {
     fn from(v: Vec<&'a str>) -> Self {
-        let mut offsets = vec![];
-        let mut values = vec![];
+        let mut offsets = Vec::with_capacity(v.len() + 1);
+        let mut values = Vec::with_capacity(v.len() + 1);
```

Comment:
After reviewing this again, I realize that this change is good for `offsets` but maybe not for `values`. We can only allocate `values` to the correct size if we first go and sum the lengths of all of the input strings, which would be expensive. Maybe we should just go back to `vec![]` or `Vec::new()` for this one?

Reply:
Ah, yes. Also since `v` can contain empty strings, we do not even have a lower bound for an initial capacity, so I'll revert to `Vec::new()` for `values`.